Error messages for FT/PT courses where a study mode is now full (#130)
DFE-Digital/bat-design-history: app/posts/apply-for-teacher-training/2020-03-02-full-or-unavailable-courses.md

This is a prevention technique at the most appropriate point. We can nudge users earlier on in their journey, which would mean these messages are less likely to be seen. We aren’t designing these warnings yet.

### Errors and warning messages

Course is full:

> You cannot apply to ‘[Provider] – [Course and code]’ because it has no vacancies

Course has been withdrawn:

> You cannot apply to ‘[Provider] – [Course and code]’ because it is not running

Location is full:

> Your chosen location for ‘[Provider] – [Course and code]’ has no vacancies

Study mode is full (only for courses that offer a choice between full time and part time):

> You cannot apply for a [part time|full time] course at ‘[Provider] – [Course and code]’ because it has no [part time|full time] vacancies

Location is full for a study mode (only for courses that offer a choice between full time and part time):

> Your chosen location for ‘[Provider] – [Course and code]’ has no [part time|full time] vacancies

### Actions available

Each error state has these options for recovery:

- deleting the choice
- changing to another course
- contacting the training provider to discuss options (see below)

> When a location is full, there is another choice, the first option is to ‘pick a new location’.

Reviewer: ```suggestion
When a location is full and another location is available, the first option is to ‘pick a new location’.
```
